core/src/components/toast/toast.md.vars.scss

```diff
@@ -4,91 +4,92 @@
+$toast-md-box-shadow: 0 3px 5px -1px rgba(0, 0, 0, 0.2), 0 6px 10px 0 rgba(0, 0, 0, 0.14),
+  0 1px 18px 0 rgba(0, 0, 0, 0.12) !default;
 
 /// @prop - Font size of the toast
-$toast-md-font-size:                               14px !default;
+$toast-md-font-size: dynamic-font(14px) !default;
 
 /// @prop - Color of the toast
-$toast-md-color:                                   $background-color-step-50 !default;
+$toast-md-color: $background-color-step-50 !default;
 
 /// @prop - Border radius of the toast wrapper
-$toast-md-border-radius:                           4px !default;
+$toast-md-border-radius: 4px !default;
 
 /// @prop - Font size of the toast message
-$toast-md-header-line-height:                      20px !default;
+$toast-md-header-line-height: dynamic-type(20px) !default;
 
 /// @prop - Font size of the toast message
-$toast-md-header-font-weight:                      500 !default;
+$toast-md-header-font-weight: 500 !default;
 
 /// @prop - Spacing between the header and the message
-$toast-md-header-margin-bottom:                    2px !default;
+$toast-md-header-margin-bottom: 2px !default;
 
 /// @prop - Font size of the toast message
-$toast-md-message-line-height:                     20px !default;
+$toast-md-message-line-height: dynamic-type(20px) !default;
 
 /// @prop - Padding top of the toast message
-$toast-md-content-padding-top:                     14px !default;
+$toast-md-content-padding-top: 14px !default;
 
 /// @prop - Padding end of the toast content
-$toast-md-content-padding-end:                     16px !default;
+$toast-md-content-padding-end: 16px !default;
 
 /// @prop - Padding bottom of the toast content
-$toast-md-content-padding-bottom:                  $toast-md-content-padding-top !default;
+$toast-md-content-padding-bottom: $toast-md-content-padding-top !default;
 
 /// @prop - Padding start of the toast content
-$toast-md-content-padding-start:                   $toast-md-content-padding-end !default;
+$toast-md-content-padding-start: $toast-md-content-padding-end !default;
 
 /// @prop - Padding top of the toast button
-$toast-md-button-padding-top:                      10px !default;
+$toast-md-button-padding-top: 10px !default;
 
 /// @prop - Padding end of the toast button
-$toast-md-button-padding-end:                      15px !default;
+$toast-md-button-padding-end: 15px !default;
 
 /// @prop - Padding bottom of the toast button
-$toast-md-button-padding-bottom:                   $toast-md-button-padding-top !default;
+$toast-md-button-padding-bottom: $toast-md-button-padding-top !default;
 
 /// @prop - Padding start of the toast button
-$toast-md-button-padding-start:                    15px !default;
+$toast-md-button-padding-start: 15px !default;
 
 /// @prop - Font size of the toast button
-$toast-md-button-font-size:                        14px !default;
+$toast-md-button-font-size: dynamic-font-clamp(1, 14px, 1.294) !default;
```

Comment:
Clamping was used to prevent the button from scaling and overflowing the container.

|With dynamic font|With dynamic font and clamping|
|----|----|
|![CleanShot 2023-08-28 at 11 12 57](https://github.com/ionic-team/ionic-framework/assets/13732623/6a214d84-6fe0-4f35-b6fd-71bd87a08ace)|![CleanShot 2023-08-28 at 11 11 17](https://github.com/ionic-team/ionic-framework/assets/13732623/0982c92d-2274-4880-bde8-d1fee8f1c36a)|

Reply:
I thought about this some more. Both Material Design 2 and Material Design 3 only recommend text up to two lines:

**Material Design 2**

<img width="802" alt="Screenshot 2023-08-30 at 4 45 41 PM" src="https://github.com/ionic-team/ionic-framework/assets/6577830/8d0429ed-bcbd-4a2b-a510-01a3d77a96bb">

**Material Design 3**

<img width="990" alt="Screenshot 2023-08-30 at 4 47 19 PM" src="https://github.com/ionic-team/ionic-framework/assets/6577830/11c602a1-0ba1-4742-a81b-6c58eef3678a">

If we cut the example down to a two-line example (on a Pixel 5) with `min-width: 0` on the content it still looks fine at 310%:

| `100%` | `310%` |
| ---| ---|
| ![100%(Pixel 5)](https://github.com/ionic-team/ionic-framework/assets/6577830/f3e34ece-5959-4ca3-b8e9-39ae26315725) | ![310%(Pixel 5)](https://github.com/ionic-team/ionic-framework/assets/6577830/2fc9e5fd-91b8-47ee-b2a3-739aec0c732e) |

Thoughts? I just want to avoid any patterns that go against MD spec.

Reply:
I like that a lot! Thanks for the extra investigation, I'll apply this 👍